sparta/scripts/GenRegisterJSON.py

```diff
@@ -0,0 +1,498 @@
+"""
+
+CSR_NUM_FILE_FOOTER = """
+}
+"""
+
+class GenRegisterJSON():
+    """Generates register definition JSON files.
+
+    Args:
+        group_name (RegisterGroup): Name of register group.
+        num_regs (int): Number of registers in the group.
+        reg_size (int): Size of the registers in bytes (must be power of 2)
+    """
+
+    def __init__(self, group_num, num_regs, reg_size):
+        self.group_num = int(group_num)
+        self.group_name = GetGroupName(group_num)
+        self.num_regs = num_regs
+        assert math.log(reg_size, 2).is_integer(), 'reg_size must be a power of 2!'
+        self.reg_size = reg_size
+        self.reg_defs = []
+
+        self.readonly_fields = {}
+        self.readonly_fields["b31_00"] = {"desc":     "read-only",
+                                          "low_bit":  0,
+                                          "high_bit": 31,
+                                          "readonly": True}
+        self.readonly_fields["b63_00"] = {"desc":     "read-only",
+                                          "low_bit":  0,
+                                          "high_bit": 63,
+                                          "readonly": True}
+
+        for num in range(0, self.num_regs):
+            name = self.get_int_reg_name(num)
+        if group_num is RegisterGroup.INT:
+            self.gen_int_reg_defs()
+        elif group_num is RegisterGroup.FP:
+            self.gen_fp_reg_defs()
+        elif group_num is RegisterGroup.CSR:
+            self.gen_csr_reg_defs()
+            self.gen_csr_header_file()
+            self.gen_csr_bit_fields()
+        elif group_num is RegisterGroup.VEC:
+            self.gen_vec_reg_defs()
+
+    def get_int_reg_name(self, num):
+        """Returns int register name from register number"""
+        return "x"+str(num)
+
+    def get_int_reg_desc(self, num):
+        """Returns int register description from register number"""
+        return "int register "+str(num)
+
+    def gen_int_reg_defs(self):
+        """Generates integer register definitions
+        """
+        for num in range(0, self.num_regs):
+            name = self.get_int_reg_name(num)
+            alias = INT_ALIASES[name] if (name in INT_ALIASES.keys()) else []
+            fields = {}
+            if (num == 0):
+                if (self.reg_size == 4):
+                    fields["b31_00"] = self.readonly_fields["b31_00"]
+                else:
+                    fields["b63_00"] = self.readonly_fields["b63_00"]
+
+            self.reg_defs.append(self.__CreateRegDict({
+                "name":          name,
+                "num":           num,
+                "desc":          self.get_int_reg_desc(num),
+                "size":          self.reg_size,
+                "aliases":       alias,
+                "fields":        fields,
+                "initial_value": 0,
+                "enabled":       True}))
+
+        #Include PC, Athena expects the PC to always be 64-bits
```

Comment:
Can you delete lines 243-265? They don't need to be checked into Sparta.

Reply:
Done (including removing resv_valid)